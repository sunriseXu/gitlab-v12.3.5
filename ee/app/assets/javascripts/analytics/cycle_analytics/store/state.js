import { DEFAULT_DATA_TIME_FRAME } from '../constants';

export default () => ({
  endpoints: {
    cycleAnalyticsData: '',
    stageData: '',
  },

  dataTimeframe: DEFAULT_DATA_TIME_FRAME,

  isLoading: false,
  isLoadingStage: false,

  isEmptyStage: false,

  isAddingCustomStage: false,

  selectedGroup: null,
  selectedProjectIds: [],
  selectedStageName: null,

  events: [],
  stages: [],
  summary: [],
});
