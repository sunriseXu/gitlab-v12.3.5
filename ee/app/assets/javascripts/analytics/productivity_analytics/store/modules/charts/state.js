import { chartKeys, chartTypes } from '../../../constants';

export default () => ({
  charts: {
    [chartKeys.main]: {
      isLoading: false,
      hasError: false,
      data: {},
      selected: [],
      params: {
        chartType: chartTypes.histogram,
      },
    },
    [chartKeys.timeBasedHistogram]: {
      isLoading: false,
      hasError: false,
      data: {},
      selected: [],
      params: {
        metricType: 'time_to_first_comment',
        chartType: chartTypes.histogram,
      },
    },
    [chartKeys.commitBasedHistogram]: {
      isLoading: false,
      hasError: false,
      data: {},
      selected: [],
      params: {
        metricType: 'commits_count',
        chartType: chartTypes.histogram,
      },
    },
    [chartKeys.scatterplot]: {
      isLoading: false,
      hasError: false,
      data: {},
      selected: [],
      params: {
        chartType: chartTypes.scatterplot,
      },
    },
  },
});
