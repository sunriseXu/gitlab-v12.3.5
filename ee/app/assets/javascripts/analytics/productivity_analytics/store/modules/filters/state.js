export default () => ({
  groupNamespace: null,
  projectPath: null,
  filters: '',
  daysInPast: 90,
});
