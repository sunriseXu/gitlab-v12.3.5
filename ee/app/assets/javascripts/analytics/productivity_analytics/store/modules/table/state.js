import { tableSortOrder } from '../../../constants';

export default () => ({
  isLoadingTable: false,
  hasError: false,
  mergeRequests: [],
  pageInfo: {},
  sortOrder: tableSortOrder.asc.value,
  sortField: 'time_to_merge',
  columnMetric: 'time_to_first_comment',
});
