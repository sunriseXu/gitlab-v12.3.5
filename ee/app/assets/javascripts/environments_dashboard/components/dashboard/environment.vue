<script>
import _ from 'underscore';
import { GlTooltipDirective, GlLink } from '@gitlab/ui';
import { s__, __, sprintf } from '~/locale';
import Icon from '~/vue_shared/components/icon.vue';
import timeagoMixin from '~/vue_shared/mixins/timeago';
import UserAvatarLink from '~/vue_shared/components/user_avatar/user_avatar_link.vue';
import Commit from '~/vue_shared/components/commit.vue';
import Alerts from 'ee/vue_shared/dashboards/components/alerts.vue';
import TimeAgo from 'ee/vue_shared/dashboards/components/time_ago.vue';
import { STATUS_FAILED, STATUS_RUNNING } from 'ee/vue_shared/dashboards/constants';
import ProjectPipeline from 'ee/vue_shared/dashboards/components/project_pipeline.vue';
import EnvironmentHeader from './environment_header.vue';

export default {
  components: {
    EnvironmentHeader,
    UserAvatarLink,
    GlLink,
    Commit,
    Alerts,
    ProjectPipeline,
    TimeAgo,
    Icon,
  },
  directives: {
    'gl-tooltip': GlTooltipDirective,
  },
  mixins: [timeagoMixin],
  props: {
    environment: {
      type: Object,
      required: true,
    },
  },
  tooltips: {
    timeAgo: __('Finished'),
    triggerer: __('Triggerer'),
    job: s__('EnvironmentsDashboard|Job: %{job}'),
  },
  noDeploymentMessage: __('This environment has no deployments yet.'),
  computed: {
    hasPipelineFailed() {
      return (
        this.lastPipeline &&
        this.lastPipeline.details &&
        this.lastPipeline.details.status &&
        this.lastPipeline.details.status.group === STATUS_FAILED
      );
    },
    hasPipelineErrors() {
      return this.environment.alert_count > 0;
    },
    cardClasses() {
      return {
        'dashboard-card-body-warning': !this.hasPipelineFailed && this.hasPipelineErrors,
        'dashboard-card-body-failed': this.hasPipelineFailed,
        'bg-secondary': !this.hasPipelineFailed && !this.hasPipelineErrors,
      };
    },
    user() {
      return this.lastDeployment && !_.isEmpty(this.lastDeployment.user)
        ? this.lastDeployment.user
        : null;
    },
    lastPipeline() {
      return !_.isEmpty(this.environment.last_pipeline) ? this.environment.last_pipeline : null;
    },
    lastDeployment() {
      return !_.isEmpty(this.environment.last_deployment) ? this.environment.last_deployment : null;
    },
    deployable() {
      return !_.isEmpty(this.lastDeployment.deployable) ? this.lastDeployment.deployable : null;
    },
    commit() {
      return !_.isEmpty(this.lastDeployment.commit) ? this.lastDeployment.commit : {};
    },
    jobTooltip() {
      return sprintf(this.$options.tooltips.job, { job: this.buildName });
    },
    commitRef() {
      return this.lastDeployment && !_.isEmpty(this.lastDeployment.commit)
        ? {
            ...this.lastDeployment.commit,
            ...this.lastDeployment.ref,
            ref_url: this.lastDeployment.ref.ref_path,
          }
        : {};
    },
    finishedTime() {
      return this.deployable.updated_at;
    },
    finishedTimeTitle() {
      return this.tooltipTitle(this.finishedTime);
    },
    shouldShowTimeAgo() {
      return (
        this.deployable &&
        this.deployable.status &&
        this.deployable.status.group !== STATUS_RUNNING &&
        this.finishedTime
      );
    },
    buildName() {
      if (
        this.environment &&
        this.environment.last_deployment &&
        this.environment.last_deployment.deployable
      ) {
        const { deployable } = this.environment.last_deployment;
        return `${deployable.name} #${deployable.id}`;
      }
      return '';
    },
  },
};
</script>
<template>
  <div class="dashboard-card card border-0">
    <environment-header
      :environment="environment"
      :has-pipeline-failed="hasPipelineFailed"
      :has-errors="hasPipelineErrors"
    />

    <div :class="cardClasses" class="dashboard-card-body card-body">
      <div v-if="lastDeployment" class="row">
        <div class="col-1 align-self-center">
          <user-avatar-link
            v-if="user"
            :link-href="user.path"
            :img-src="user.avatar_url"
            :tooltip-text="user.name"
            :img-size="32"
          />
        </div>

        <div class="col-10 col-sm-6 pr-0 pl-5 align-self-center align-middle ci-table">
          <div class="branch-commit">
            <icon name="work" />
            <gl-link v-gl-tooltip="jobTooltip" :href="deployable.build_path" class="str-truncated">
              {{ buildName }}
            </gl-link>
          </div>
          <commit
            :tag="commitRef.tag"
            :commit-ref="commitRef"
            :short-sha="commit.short_id"
            :commit-url="commit.commit_url"
            :title="commit.title"
            :author="commit.author"
            :show-branch="true"
          />
        </div>

        <div class="col-sm-5 pl-0 text-right align-self-center d-none d-sm-block">
          <time-ago
            v-if="shouldShowTimeAgo"
            :time="finishedTime"
            :tooltip-text="$options.tooltips.timeAgo"
          />
          <alerts :count="environment.alert_count" :last-alert="environment.last_alert" />
        </div>

        <div v-if="lastPipeline" class="col-12">
          <project-pipeline
            :project-name="environment.name_with_namespace"
            :last-pipeline="lastPipeline"
            :has-pipeline-failed="hasPipelineFailed"
          />
        </div>
      </div>

      <div v-else class="h-100 d-flex justify-content-center align-items-center">
        <div class="text-plain text-metric text-center bold w-75">
          {{ $options.noDeploymentMessage }}
        </div>
      </div>
    </div>
  </div>
</template>
