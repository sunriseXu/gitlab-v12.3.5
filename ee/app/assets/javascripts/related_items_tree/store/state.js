export default () => ({
  // Initial Data
  parentItem: {},
  epicsEndpoint: '',
  issuesEndpoint: '',

  children: {},
  childrenFlags: {},
  epicsCount: 0,
  issuesCount: 0,

  // Add Item Form Data
  actionType: '',
  itemInputValue: '',
  pendingReferences: [],
  itemAutoCompleteSources: {},

  // UI Flags
  itemsFetchInProgress: false,
  itemsFetchFailure: false,
  itemsFetchResultEmpty: false,
  itemAddInProgress: false,
  itemCreateInProgress: false,
  showAddItemForm: false,
  showCreateItemForm: false,
  autoCompleteEpics: false,
  autoCompleteIssues: false,
  removeItemModalProps: {
    parentItem: {},
    item: {},
  },
});
