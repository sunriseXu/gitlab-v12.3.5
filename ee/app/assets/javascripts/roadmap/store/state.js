export default () => ({
  // API Calls
  basePath: '',
  epicsState: '',
  filterQueryString: '',
  initialEpicsPath: '',

  // Data
  epics: [],
  epicIds: [],
  currentGroupId: -1,
  timeframe: [],
  extendedTimeframe: [],
  presetType: '',
  sortedBy: '',

  // UI Flags
  defaultInnerHeight: 0,
  isChildEpics: false,
  windowResizeInProgress: false,
  epicsFetchInProgress: false,
  epicsFetchForTimeframeInProgress: false,
  epicsFetchFailure: false,
  epicsFetchResultEmpty: false,
});
