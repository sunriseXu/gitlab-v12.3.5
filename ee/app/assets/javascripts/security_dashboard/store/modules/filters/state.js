import { SEVERITY_LEVELS, CONFIDENCE_LEVELS, REPORT_TYPES, BASE_FILTERS } from './constants';
import { s__ } from '~/locale';

const optionsObjectToArray = obj => Object.entries(obj).map(([id, name]) => ({ id, name }));

export default () => ({
  filters: [
    {
      name: s__('SecurityDashboard|Severity'),
      id: 'severity',
      options: [BASE_FILTERS.severity, ...optionsObjectToArray(SEVERITY_LEVELS)],
      hidden: false,
      selection: new Set(['all']),
    },
    {
      name: s__('SecurityDashboard|Confidence'),
      id: 'confidence',
      options: [BASE_FILTERS.confidence, ...optionsObjectToArray(CONFIDENCE_LEVELS)],
      hidden: false,
      selection: new Set(['all']),
    },
    {
      name: s__('SecurityDashboard|Report type'),
      id: 'report_type',
      options: [BASE_FILTERS.report_type, ...optionsObjectToArray(REPORT_TYPES)],
      hidden: false,
      selection: new Set(['all']),
    },
    {
      name: s__('SecurityDashboard|Project'),
      id: 'project_id',
      options: [BASE_FILTERS.project_id],
      hidden: false,
      selection: new Set(['all']),
    },
  ],
  hide_dismissed: true,
});
