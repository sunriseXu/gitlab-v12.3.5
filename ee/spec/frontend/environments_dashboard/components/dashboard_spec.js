import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import { GlButton, GlModal } from '@gitlab/ui';
import ProjectSelector from '~/vue_shared/components/project_selector/project_selector.vue';
import createStore from 'ee/vue_shared/dashboards/store/index';
import state from 'ee/vue_shared/dashboards/store/state';
import component from 'ee/environments_dashboard/components/dashboard/dashboard.vue';
import ProjectHeader from 'ee/environments_dashboard/components/dashboard/project_header.vue';
import Environment from 'ee/environments_dashboard/components/dashboard/environment.vue';

import environment from './mock_environment.json';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('dashboard', () => {
  const Component = localVue.extend(component);
  let actionSpies;
  const store = createStore();
  let wrapper;
  let propsData;

  beforeEach(() => {
    actionSpies = {
      addProjectsToDashboard: jest.fn(),
      clearSearchResults: jest.fn(),
      setSearchQuery: jest.fn(),
      fetchSearchResults: jest.fn(),
      removeProject: jest.fn(),
      toggleSelectedProject: jest.fn(),
    };
    propsData = {
      addPath: 'mock-addPath',
      listPath: 'mock-listPath',
      emptyDashboardSvgPath: '/assets/illustrations/operations-dashboard_empty.svg',
      emptyDashboardHelpPath: '/help/user/operations_dashboard/index.html',
    };

    wrapper = shallowMount(Component, {
      propsData,
      localVue,
      store,
      methods: {
        fetchProjects: () => {},
        ...actionSpies,
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
    store.replaceState(state());
  });

  it('should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders the dashboard title', () => {
    expect(wrapper.find('.js-dashboard-title').text()).toBe('Environments Dashboard');
  });

  describe('add projects button', () => {
    let button;

    beforeEach(() => {
      button = wrapper.find(GlButton);
    });

    it('is labelled correctly', () => {
      expect(button.text()).toBe('Add projects');
    });
  });

  describe('wrapped components', () => {
    beforeEach(() => {
      store.state.projects = [
        {
          id: 0,
          name: 'test',
          namespace: { name: 'test', id: 0 },
          environments: [{ ...environment, id: 0 }, environment],
        },
        { id: 1, name: 'test', namespace: { name: 'test', id: 0 }, environments: [environment] },
      ];
    });

    describe('project header', () => {
      it('should have one project header per project', () => {
        const headers = wrapper.findAll(ProjectHeader);
        expect(headers.length).toBe(2);
      });

      it('should remove a project if it emits `remove`', () => {
        const header = wrapper.find(ProjectHeader);
        header.vm.$emit('remove');
        expect(actionSpies.removeProject).toHaveBeenCalled();
      });
    });

    describe('environment component', () => {
      it('should have one environment component per environment', () => {
        const environments = wrapper.findAll(Environment);
        expect(environments.length).toBe(3);
      });
    });

    describe('project selector modal', () => {
      beforeEach(() => {
        wrapper.find(GlButton).trigger('click');
      });

      it('should fire the add projects action on ok', () => {
        wrapper.find(GlModal).vm.$emit('ok');
        expect(actionSpies.addProjectsToDashboard).toHaveBeenCalled();
      });

      it('should fire clear search when the modal is hidden', () => {
        wrapper.find(GlModal).vm.$emit('hidden');
        expect(actionSpies.clearSearchResults).toHaveBeenCalled();
      });

      it('should set the search query when searching', () => {
        wrapper.find(ProjectSelector).vm.$emit('searched', 'test');
        expect(actionSpies.setSearchQuery).toHaveBeenCalledWith('test');
      });

      it('should fetch query results when searching', () => {
        wrapper.find(ProjectSelector).vm.$emit('searched', 'test');
        expect(actionSpies.fetchSearchResults).toHaveBeenCalled();
      });

      it('should toggle a project when clicked', () => {
        wrapper.find(ProjectSelector).vm.$emit('projectClicked', { name: 'test', id: 1 });
        expect(actionSpies.toggleSelectedProject).toHaveBeenCalledWith({ name: 'test', id: 1 });
      });
    });
  });
});
