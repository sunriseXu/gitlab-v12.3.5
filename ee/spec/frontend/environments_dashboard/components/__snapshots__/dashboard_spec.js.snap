// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard should match the snapshot 1`] = `
<div
  class="operations-dashboard"
>
  <glmodal-stub
    modalclass=""
    modalid="add-projects-modal"
    ok-disabled="true"
    ok-title="Add projects"
    ok-variant="success"
    title="Add projects"
    titletag="h4"
  >
    <projectselector-stub
      projectsearchresults=""
      selectedprojects=""
    />
  </glmodal-stub>
   
  <div
    class="page-title-holder flex-fill d-flex align-items-center"
  >
    <h1
      class="js-dashboard-title page-title text-nowrap flex-fill"
    >
      
      Environments Dashboard
    
    </h1>
     
    <glbutton-stub
      class="js-add-projects-button btn btn-success"
      role="button"
    >
      
      Add projects
    
    </glbutton-stub>
  </div>
   
  <div
    class="prepend-top-default"
  >
    <div
      class="row prepend-top-20 text-center"
    >
      <div
        class="col-12 d-flex justify-content-center svg-content"
      >
        <img
          class="js-empty-state-svg col-12 prepend-top-20"
          src="/assets/illustrations/operations-dashboard_empty.svg"
        />
      </div>
       
      <h4
        class="js-title col-12 prepend-top-20"
      >
        
        Add a project to the dashboard
      
      </h4>
       
      <div
        class="col-12 d-flex justify-content-center"
      >
        <span
          class="js-sub-title mw-460 text-tertiary text-left"
        >
          
          The environments dashboard provides a summary of each project's environments' status, including pipeline and alert statuses.
        
        </span>
      </div>
       
      <div
        class="col-12"
      >
        <a
          class="js-documentation-link btn btn-primary prepend-top-default append-bottom-default"
          href="/help/user/operations_dashboard/index.html"
        >
          
          View documentation
        
        </a>
      </div>
    </div>
  </div>
</div>
`;
